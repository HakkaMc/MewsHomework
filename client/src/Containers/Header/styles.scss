.gridContainer {
  align-items: center;
  justify-content: space-between;
  padding: 20px 0;
}

.toolbar {
  padding: 0;
}

.search {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;
  color: white;
  background-color: rgba(white, 0.15);
  padding: 5px 10px;
  border-radius: 3px;
  transition: background-color 0.2s linear 0s;

  @media all and (max-width: 599px) {
    margin-top: 10px;
  }

  &:hover {
    background-color: rgba(white, 0.25);
  }

  & > *:first-child {
    margin-right: 5px;
  }

  input {
    color: white;
  }

  & > *:last-child {
    margin-left: 5px;
    cursor: pointer;
  }
}
