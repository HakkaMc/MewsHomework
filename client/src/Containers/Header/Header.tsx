import React from "react";
import { connect } from "react-redux";
import { change } from "redux-form";
import { compose } from "recompose";
import {
	FormattedMessage,
	injectIntl,
	WrappedComponentProps as IntlProps
} from "react-intl";
import {
	AppBar,
	Toolbar,
	Typography,
	Grid,
	Container
} from "@material-ui/core";
import SearchIcon from "@material-ui/icons/Search";
import ClearIcon from "@material-ui/icons/Clear";

import { actions } from "../../reducers";
import Search from "../../Components/Search/Search";
import { messages } from "../../constants/locales/messages";

import styles from "./styles.scss";

export interface Props extends IntlProps {
	change: typeof change;
	clearAll: typeof actions.movies.clearAll;
}

export const Header: React.FC<Props> = (props: Props) => {
	const clearSearch = () => {
		props.change("search", "movieName", "");
		props.clearAll();
	};

	return (
		<AppBar position="static">
			<Container>
				<Toolbar variant="dense" disableGutters>
					<Grid container className={styles.gridContainer}>
						<Grid item>
							<Typography variant="h4">
								<FormattedMessage {...messages.Header.title} />
							</Typography>
						</Grid>
						<Grid item xs={12} sm={5} md={"auto"}>
							<label className={styles.search}>
								<SearchIcon />
								<Search
									light
									placeholder={props.intl.formatMessage(
										messages.Home.placeholder
									)}
								/>
								<ClearIcon className={styles.clearIcon} onClick={clearSearch} />
							</label>
						</Grid>
					</Grid>
				</Toolbar>
			</Container>
		</AppBar>
	);
};

export default compose<Props, any>(
	injectIntl,
	connect(null, {
		change: change,
		clearAll: actions.movies.clearAll
	})
)(Header);
