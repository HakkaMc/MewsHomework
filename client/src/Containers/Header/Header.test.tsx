import React from "react";
import { shallow } from "enzyme";
import ClearIcon from "@material-ui/icons/Clear";

import { Header, Props } from "./Header";

describe("<Header/>", () => {
	let props: Props;

	beforeEach(() => {
		const intl: any = {
			formatMessage: jest.fn()
		};

		props = {
			change: jest.fn(),
			clearAll: jest.fn(),
			intl
		};
	});

	it("Match snapshot", () => {
		const wrapper = shallow(<Header {...props} />);
		expect(wrapper).toMatchSnapshot();
	});

	it("Clear icon click calls props.change and props.clearAll functions", () => {
		const wrapper = shallow(<Header {...props} />);
		wrapper.find(ClearIcon).simulate("click");
		expect(props.change).toHaveBeenCalled();
		expect(props.clearAll).toHaveBeenCalled();
	});
});
