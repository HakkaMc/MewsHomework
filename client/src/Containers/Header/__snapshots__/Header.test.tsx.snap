// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Match snapshot 1`] = `
<WithStyles(ForwardRef(AppBar))
  position="static"
>
  <WithStyles(ForwardRef(Container))>
    <WithStyles(ForwardRef(Toolbar))
      disableGutters={true}
      variant="dense"
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Typography))
            variant="h4"
          >
            <FormattedMessage
              defaultMessage="Movie database"
              id="Header.title"
              values={Object {}}
            />
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          md="auto"
          sm={5}
          xs={12}
        >
          <label
            className={[Function]}
          >
            <SearchIcon />
            <ReduxForm
              destroyOnUnmount={false}
              enableReinitialize={false}
              forceUnregisterOnUnmount={false}
              form="search"
              getFormState={[Function]}
              keepDirtyOnReinitialize={false}
              light={true}
              persistentSubmitErrors={false}
              pure={true}
              shouldAsyncValidate={[Function]}
              shouldError={[Function]}
              shouldValidate={[Function]}
              shouldWarn={[Function]}
              submitAsSideEffect={false}
              touchOnBlur={true}
              touchOnChange={false}
              updateUnregisteredFields={false}
            />
            <ClearIcon
              onClick={[Function]}
            />
          </label>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(Container))>
</WithStyles(ForwardRef(AppBar))>
`;
